### ISSUE TITLE
Statevector.from_instruction with controlled opaque gates breaks

### ISSUE THREAD

**Issue by Cryoris** [Qiskit/qiskit-terra#4681]
### Information

- **Qiskit Terra version**: 0.16.0 (stable)
- **Python version**: 3.7.7 (also on 3.8.3)
- **Operating system**: macOS Catalina

### What is the current behavior?

Simulating a circuit containing a controlled circuit with the `Statevector` class yields all zero amplitudes.
```python
x = QuantumCircuit(1)
x.x(0)
circ = QuantumCircuit(2)
circ.append(x.to_gate().control(), [0, 1])
print(Statevector.from_instruction(circ))  # [0 0 0 0]
print(execute(circ, Aer.get_backend('statevector_simulator')).result().get_statevector())  # [1 0 0 0]
```
This is the case for different circuits, not just with a single X gate. Not calling `control` on the gate works correctly.

**Comment by obliviateandsurrender** [comment]
This gives me the correct result. Using Python 3.8 and qiskit 0.19.6 

![image](https://user-images.githubusercontent.com/23234279/87704433-03cf8e80-c7ba-11ea-9d66-3d339d11ebcc.png)

**Comment by Cryoris** [comment]
Hm, maybe this worked earlier, now it doesn't give zero amplitudes but breaks. I think this is a know bug with the quantum info due to nested quantum registers.
```python
>>> from qiskit import QuantumCircuit
>>> from qiskit.quantum_info import Statevector
>>> x = QuantumCircuit(1)
>>> x.x(0)
>>> circ = QuantumCircuit(2)
>>> circ.append(x.to_gate().control(), [0, 1])
>>> print(Statevector.from_instruction(circ))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/jul/Work/Qiskit/qiskit-terra/qiskit/quantum_info/states/statevector.py", line 529, in from_instruction
    return Statevector._evolve_instruction(vec, instruction)
  File "/Users/jul/Work/Qiskit/qiskit-terra/qiskit/quantum_info/states/statevector.py", line 690, in _evolve_instruction
    Statevector._evolve_instruction(statevec, instr, qargs=new_qargs)
  File "/Users/jul/Work/Qiskit/qiskit-terra/qiskit/quantum_info/states/statevector.py", line 690, in _evolve_instruction
    Statevector._evolve_instruction(statevec, instr, qargs=new_qargs)
  File "/Users/jul/Work/Qiskit/qiskit-terra/qiskit/quantum_info/states/statevector.py", line 661, in _evolve_instruction
    return Statevector._evolve_operator(statevec, Operator(mat), qargs=qargs)
  File "/Users/jul/Work/Qiskit/qiskit-terra/qiskit/quantum_info/states/statevector.py", line 642, in _evolve_operator
    np.reshape(statevec.data, pre_tensor_shape), axes), contract_shape)
  File "<__array_function__ internals>", line 6, in transpose
  File "/Users/jul/anaconda3/envs/py377/lib/python3.7/site-packages/numpy/core/fromnumeric.py", line 651, in transpose
    return _wrapfunc(a, 'transpose', axes)
  File "/Users/jul/anaconda3/envs/py377/lib/python3.7/site-packages/numpy/core/fromnumeric.py", line 61, in _wrapfunc
    return bound(*args, **kwds)
ValueError: axes don't match array
>>>
```

**Comment by kdk** [comment]
Closing as resolved by #5286 .
